src/ecalc/libraries/libecalc/common/libecalc/input/yaml_types/components/generator_set.py

```diff
@@ -0,0 +1,50 @@
+from typing import Any, Dict, List, Type
+
+from libecalc.dto.base import ConsumerUserDefinedCategoryType
```

Comment:
Just want to say that I agree on this (I think), that the yaml-models depend on DTO (the common input layer that we have now...), instead of replicating it.